### ISSUE TITLE
Serializing graphs using acid-state

### ISSUE THREAD

**Issue by srid** [snowleopard/alga#176]
I'm using `acid-state` to serialize graphs created using this library. acid-state uses the `cereal` package which in turns requires types be an instance of `Generic`. However I can't derive a Generic instance for `AdjacencyMap` because the `AM` constructor is not exposed:

```haskell
    • Can't make a derived instance of ‘Generic (AdjacencyMap v)’:
        The data constructors of ‘AdjacencyMap’ are not all in scope
          so you cannot derive an instance for it
    • In the stand-alone deriving instance for
        ‘Generic v => Generic (AdjacencyMap v)’
   |
35 | deriving instance Generic v => Generic (AdjacencyMap v)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

How would you suggest that I proceed from here? 

Thanks.

**Comment by snowleopard** [comment]
Hi @srid! As a quick solution, you could import the constructor from the [internal module](https://hackage.haskell.org/package/algebraic-graphs-0.3/docs/Algebra-Graph-AdjacencyMap-Internal.html).

A more long term solution would be to add a `Generic` instance for `AdjacencyMap`. Would you like to do that? I'm happy to review a PR. It's not a difficult change, so I'm happy to do this myself too.

**Comment by srid** [comment]
> Would you like to do that?

Absolutely, I'll take a stab at it. :-)

**Comment by srid** [comment]
Fixed in #178
